operations/mimir/ingester-automated-downscale-v2.libsonnet

```diff
@@ -0,0 +1,99 @@
+{
+  // Allow ingesters to be safely downscaled using the "prepare-downscale" and "resource-mirroring" features
+  // of the rollout-operator. Only available when using multi-zone ingesters. This feature can be enabled
+  // via the ingester_automated_downscale_enabled flag.
+  //
+  // When enabled, ReplicaTemplate/ingester-zone-a object is installed into the namespace. This object holds desired
+  // number of replicas for each ingester zone in spec.replicas field. (There is only one ReplicaTemplate object,
+  // and all ingester zones use it.)
+  //
+  // Scaledown is initiated by decreasing spec.replicas field of ReplicaTemplate/ingester-zone-a object.
+  // Rollout-operator will notice the change, set ingesters that are going to be downscaled (in all zones)
+  // to read-only mode, and wait until ingesters were in read-only mode for pre-configured time
+  // (ingester_automated_downscale_delay).
+  //
+  // After this time is reached, rollout-operator will finally adjust number of replicas in ingester StatefulSets
+  // (for all zones).
+  //
+  // The number of replicas is still controlled by $._config.multi_zone_ingester_replicas until a reasonable metric to scale it
+  // automatically is determined and tested.
+  _config+: {
+    ingester_automated_downscale_v2_enabled: false,
+
+    // Allow to selectively enable it on a per-zone basis. Note that zone-a must be enabled first.
+    ingester_automated_downscale_v2_zone_a_enabled: $._config.ingester_automated_downscale_v2_enabled,
```

Comment:
Do we want to keep this per-zone setting, or simplify it down to a single flag?

Reply:
Let's simplify.